README.md
# WordPress Starter Premium plugin for WordPress

[![Codeship status](https://www.codeship.io/projects/TBD/status)](https://www.codeship.io/projects/TBD)

## Welcome to the WordPress Starter Premium Github repository

WordPress Starter Premium plugin allows you to add custom fields to the edit screen bulk and quick edit panels.

While the documentation for the [WordPress Starter Premium plugin](http://wordpress.org/plugins/wordpress-starter/) can be found at WordPress.org Plugin Directory, here you can browse the source of the project, find and discuss open issues and, even [contribute yourself](https://github.com/michael-cannon/wordpress-starter/blob/master/CONTRIBUTING.md).

## Installation

Here's a [guide on how to install WordPress Starter Premium in your WordPress site](http://wordpress.org/plugins/wordpress-starter/installation/). If you want to run the Git version though, you have two options:

* You can clone the GitHub repository: https://github.com/michael-cannon/wordpress-starter.git
* Download it directly as a ZIP file: https://github.com/michael-cannon/wordpress-starter/archive/master.zip

This will download the latest development version of WordPress Starter Premium. While this version is usually stable, it is not recommended for use in a production environment.

## Bugs

If you find an issue, [let us know here](https://github.com/michael-cannon/wordpress-starter/issues/new)!

## Support

This is a developer's portal for WordPress Starter Premium and should not be used for support. Please visit the [WordPress Starter Premium Knowledge Base](https://nodedesk.zendesk.com/hc/en-us/sections/200861112) for frequently asked questions, offering ideas, or getting support.

## Contributions

Anyone is welcome to contribute to WordPress Starter Premium. Please [read the guidelines](https://github.com/michael-cannon/wordpress-starter/blob/master/CONTRIBUTING.md) for contributing to this repository.

There are various ways you can contribute:

* [Raise an issue](https://github.com/michael-cannon/wordpress-starter/issues) on GitHub.
* Fork WordPress Starter Premium and send a Pull Request with your bug fixes and/or new features.
* [Translate WordPress Starter Premium into different languages](https://nodedesk.zendesk.com/hc/en-us/articles/202294892).
* Provide feedback and [suggestions on enhancements](https://github.com/michael-cannon/wordpress-starter/issues?direction=desc&labels=Enhancement&page=1&sort=created&state=open).

## Thank You
* Thank you [WordPress SEO by Yoast](https://github.com/jdevalk/wordpress-seo/blob/master/README.md) for this README.md and like inspirations.


CHANGELOG.md
# CHANGELOG WordPress Starter Premium

## master

## 1.0.0
* Initial code release 

## 0.0.0
* $this to __CLASS__
* Add Codeship.io configuration
* Add licensing texts
* Add missing `notice_version`
* Add plugin init load action `wpsp_init`
* Add WPS_PLUGIN_DIR path
* Add [[gallery]]
* Add `do_load` 
* Auto save options on activation and upgrades
* BUGFIX Inactive REQ_BASE due to old Aihrus Framework
* Check for active base plugin
* Check for PHP 5.3
* Correct premium prep defaults
* Enable licensing 
* Implement PHP version checking
* Implement WordPress version checking
* Include CSS in plugin prepartions
* Make scripts/styles consistent
* Move ci to tests
* Move CSS to assets
* Move files to assets and includes
* No longer require standalone free plugin activation
* Put YouTube to https
* Readme installation
* Remove unused CONSTANTS
* Require Aihrus 1.1.4
* RESOLVE Non-SSL protocol used for scripts and styles
* Revise headers
* Revise readme structure
* Revise required file paths
* Revise weekly license checking
* Separate license key entries
* Specify a “Text Domain” and “Domain Path”
* Tested up to 3.9.0
* Update Aihrus integration
* Update Axelerant store branding
* Update copyright
* Update EDD_SL_Plugin_Updater 1.1
* Update LICENSE
* Update `version_check` for free requirements
* Use $plugin_assets than $plugin_path
* Use aihr_check_aihrus_framework
* WordPress Starter Premium branding

ci/tests/bootstrap.php
<?php

require_once getenv( 'WP_TESTS_DIR' ) . '/includes/functions.php';

function _manually_load_plugin() {
	require dirname( __FILE__ ) . '/../../wordpress-starter-premium.php';
}


tests_add_filter( 'muplugins_loaded', '_manually_load_plugin' );

require getenv( 'WP_TESTS_DIR' ) . '/includes/bootstrap.php';
?>


prepare-plugin.sh
#!/bin/bash

NEW_ABBR="CBQEP_"
NEW_BASE="custom-bulk-quick-edit-premium"
NEW_CLASS="Custom_Bulk_Quick_Edit_Premium"
NEW_FILTER="${NEW_ABBR,,}"
NEW_KB_PATH="20112546"
NEW_SITE="http://aihr.us/products/${NEW_BASE}/"
NEW_SLUG="${NEW_FILTER}"
NEW_SLUG_LONG="${NEW_BASE/-/_}"
NEW_TITLE="Custom Bulk/Quick Edit Premium"
NEW_TITLE_SHORT="${NEW_TITLE/ Premium/}"

OLD_ABBR="WPSP_"
OLD_BASE="wordpress-starter-premium"
OLD_CLASS="WordPress_Starter_Premium"
OLD_FILTER="${OLD_ABBR,,}"
OLD_KB_PATH="20102742"
OLD_SITE="http://wordpress.org/plugins/${OLD_BASE}/"
OLD_SLUG="${OLD_FILTER}"
OLD_SLUG_LONG="${OLD_BASE/-/_}"
OLD_TITLE="WordPress Starter Premium"
OLD_TITLE_SHORT="${OLD_TITLE/ Premium/}"

echo
echo "Begin converting ${OLD_TITLE} to ${NEW_TITLE} plugin"

FILES=`find . -type f \( -name "*.css" -o -name "*.md" -o -name "*.php" -o -name "*.txt" -o -name "*.xml" \)`
for FILE in ${FILES} 
do
	if [[ '' != ${NEW_ABBR} ]]
	then
		perl -pi -e "s#${OLD_ABBR}#${NEW_ABBR}#g" ${FILE}
		perl -pi -e "s#${NEW_ABBR}_#${NEW_ABBR}#g" ${FILE}
	fi

	if [[ '' != ${NEW_BASE} ]]
	then
		perl -pi -e "s#${OLD_BASE}#${NEW_BASE}#g" ${FILE}
	fi

	if [[ '' != ${NEW_CLASS} ]]
	then
		perl -pi -e "s#${OLD_CLASS}#${NEW_CLASS}#g" ${FILE}
	fi

	if [[ '' != ${NEW_FILTER} ]]
	then
		perl -pi -e "s#${OLD_FILTER}#${NEW_FILTER}#g" ${FILE}
	fi

	if [[ '' != ${NEW_KB_PATH} ]]
	then
		perl -pi -e "s#${OLD_KB_PATH}#${NEW_KB_PATH}#g" ${FILE}
	fi

	if [[ '' != ${NEW_SITE} ]]
	then
		perl -pi -e "s#${OLD_SITE}#${NEW_SITE}#g" ${FILE}
	fi

	if [[ '' != ${NEW_SLUG} ]]
	then
		perl -pi -e "s#${OLD_SLUG}#${NEW_SLUG}#g" ${FILE}
		perl -pi -e "s#${NEW_SLUG}_#${NEW_SLUG}#g" ${FILE}
	fi

	if [[ '' != ${NEW_SLUG_LONG} ]]
	then
		perl -pi -e "s#${OLD_SLUG_LONG}#${NEW_SLUG_LONG}#g" ${FILE}
	fi

	if [[ '' != ${NEW_TITLE} ]]
	then
		perl -pi -e "s#${OLD_TITLE}#${NEW_TITLE}#g" ${FILE}
	fi

	if [[ '' != ${NEW_TITLE_SHORT} ]]
	then
		perl -pi -e "s#${OLD_TITLE_SHORT}#${NEW_TITLE_SHORT}#g" ${FILE}
	fi
done

if [[ -e 000-code-qa.txt ]]
then
	rm 000-code-qa.txt
fi

mv ${OLD_BASE}.php ${NEW_BASE}.php
mv assets/css/${OLD_BASE}.css assets/css/${NEW_BASE}.css
mv includes/class-${OLD_BASE}-licensing.php includes/class-${NEW_BASE}-licensing.php
mv includes/class-${OLD_BASE}.php includes/class-${NEW_BASE}.php
mv languages/${OLD_BASE}.pot languages/${NEW_BASE}.pot

if [[ -e .git ]]
then
	rm -rf .git
fi

LIB_AIHRUS="includes/libraries/aihrus"
if [[ -e ${LIB_AIHRUS} ]]
then
	rm ${LIB_AIHRUS}
fi

git init
git add *
git add .gitignore
git commit -m "Initial plugin creation"
git remote add origin git@github.com:michael-cannon/${NEW_BASE}.git
echo "git push origin master"

includes/requirements.php
<?php
/**
WordPress Starter Premium
Copyright (C) 2015 Axelerant

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License along
with this program; if not, write to the Free Software Foundation, Inc.,
51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

require_once AIHR_DIR . 'aihrus-framework.php';


function wpsp_requirements_check( $force_check = false ) {
	if ( is_plugin_active( WPSP_REQ_BASE ) ) {
		aihr_deactivate_plugin( WPSP_REQ_BASE );
		add_action( 'admin_notices', 'wpsp_notice_wps_deactivated' );
	}

	$check_okay = get_transient( 'wpsp_requirements_check' );
	if ( empty( $force_check ) && false !== $check_okay ) {
		return $check_okay;
	}

	$deactivate_reason = false;
	if ( ! aihr_check_php( WPSP_BASE, WPSP_NAME ) ) {
		$deactivate_reason = esc_html__( 'Old PHP version detected' );
	}

	if ( ! aihr_check_wp( WPSP_BASE, WPSP_NAME ) ) {
		$deactivate_reason = esc_html__( 'Old WordPress version detected' );
	}

	global $wps_activated;

	if ( empty( $wps_activated ) ) {
		$deactivate_reason = esc_html__( 'Internal WordPress Starter not detected' );
	}

	if ( ! empty( $deactivate_reason ) ) {
		aihr_deactivate_plugin( WPSP_BASE, WPSP_NAME, $deactivate_reason );
	}

	$check_okay = empty( $deactivate_reason );
	if ( $check_okay ) {
		delete_transient( 'wpsp_requirements_check' );
		set_transient( 'wpsp_requirements_check', $check_okay, HOUR_IN_SECONDS );
	}

	return $check_okay;
}


function wpsp_notice_aihrus() {
	$help_url  = esc_url( 'https://nodedesk.zendesk.com/hc/en-us/articles/202381391' );
	$help_link = sprintf( __( '<a href="%1$s">Update plugins</a>. <a href="%2$s">More information</a>.' ), self_admin_url( 'update-core.php' ), $help_url );

	$text = sprintf( esc_html__( 'Plugin "%1$s" has been deactivated as it requires a current Aihrus Framework. Once corrected, "%1$s" can be activated. %2$s' ), WPSP_NAME, $help_link );

	aihr_notice_error( $text );
}


function wpsp_notice_wps_deactivated() {
	$text = sprintf( esc_html__( 'Plugin "%1$s" has been deactivated as it is no longer required by "%2$s". You can safely delete plugin "%1$s" given that "Remove Plugin Data on Deletion?" isn\'t checked on the Reset tab of Settings.' ), WPSP_REQ_NAME, WPSP_NAME );

	aihr_notice_error( $text );
}

?>


lib/EDD_SL_Plugin_Updater.php
<?php

// uncomment this line for testing
//set_site_transient( 'update_plugins', null );

/**
 * Allows plugins to use their own update API.
 *
 * @author Pippin Williamson
 * @version 1.0
 */


class EDD_SL_Plugin_Updater {
	private $api_url  = '';
	private $api_data = array();
	private $name     = '';
	private $slug     = '';


	/**
	 * Class constructor.
	 *
	 * @uses plugin_basename()
	 * @uses hook()
	 *
	 * @param string  $_api_url     The URL pointing to the custom API endpoint.
	 * @param string  $_plugin_file Path to the plugin file.
	 * @param array   $_api_data    Optional data to send with API calls.
	 * @return void
	 */
	function __construct( $_api_url, $_plugin_file, $_api_data = null ) {
		$this->api_url  = trailingslashit( $_api_url );
		$this->api_data = urlencode_deep( $_api_data );
		$this->name     = plugin_basename( $_plugin_file );
		$this->slug     = basename( $_plugin_file, '.php');
		$this->version  = $_api_data['version'];

		// Set up hooks.
		$this->hook();
	}


	/**
	 * Set up Wordpress filters to hook into WP's update process.
	 *
	 * @uses add_filter()
	 *
	 * @return void
	 */
	private function hook() {
		add_filter( 'pre_set_site_transient_update_plugins', array( $this, 'pre_set_site_transient_update_plugins_filter' ) );
		add_filter( 'plugins_api', array( $this, 'plugins_api_filter' ), 10, 3 );
	}


	/**
	 * Check for Updates at the defined API endpoint and modify the update array.
	 *
	 * This function dives into the update api just when Wordpress creates its update array,
	 * then adds a custom API call and injects the custom plugin data retrieved from the API.
	 * It is reassembled from parts of the native Wordpress plugin update code.
	 * See wp-includes/update.php line 121 for the original wp_update_plugins() function.
	 *
	 * @uses api_request()
	 *
	 * @param array   $_transient_data Update array build by Wordpress.
	 * @return array Modified update array with custom plugin data.
	 */
	function pre_set_site_transient_update_plugins_filter( $_transient_data ) {


		if ( empty( $_transient_data ) ) return $_transient_data;

		$to_send = array( 'slug' => $this->slug );

		$api_response = $this->api_request( 'plugin_latest_version', $to_send );

		if ( false !== $api_response && is_object( $api_response ) ) {
			if ( version_compare( $this->version, $api_response->new_version, '<' ) )
				$_transient_data->response[$this->name] = $api_response;
		}
		return $_transient_data;
	}


	/**
	 * Updates information on the "View version x.x details" page with custom data.
	 *
	 * @uses api_request()
	 *
	 * @param mixed   $_data
	 * @param string  $_action
	 * @param object  $_args
	 * @return object $_data
	 */
	function plugins_api_filter( $_data, $_action = '', $_args = null ) {
		if ( ( $_action != 'plugin_information' ) || !isset( $_args->slug ) || ( $_args->slug != $this->slug ) ) return $_data;

		$to_send = array( 'slug' => $this->slug );

		$api_response = $this->api_request( 'plugin_information', $to_send );
		if ( false !== $api_response ) $_data = $api_response;

		return $_data;
	}


	/**
	 * Calls the API and, if successfull, returns the object delivered by the API.
	 *
	 * @uses get_bloginfo()
	 * @uses wp_remote_post()
	 * @uses is_wp_error()
	 *
	 * @param string  $_action The requested action.
	 * @param array   $_data   Parameters for the API action.
	 * @return false||object
	 */
	private function api_request( $_action, $_data ) {
		$data = array_merge( $this->api_data, $_data );

		if ( $data['slug'] != $this->slug )
			return;

		if ( empty( $data['license'] ) )
			return;

		$api_params = array(
			'edd_action'  => 'get_version',
			'license'   => $data['license'],
			'name'    => $data['item_name'],
			'slug'    => $this->slug,
			'author'  => $data['author']
		);
		$request = wp_remote_post( $this->api_url, array( 'timeout' => 15, 'sslverify' => false, 'body' => $api_params ) );

		if ( !is_wp_error( $request ) ):
			$request = json_decode( wp_remote_retrieve_body( $request ) );
		if ( $request )
			$request->sections = maybe_unserialize( $request->sections );
		return $request;
		else:
			return false;
		endif;
	}


}


wordpress-starter-premium.php
<?php
/**
 * Plugin Name: WordPress Starter Premium
 * Plugin URI: http://store.axelerant.com/downloads/wordpress-starter-premium/
 * Description: TBD
 * Version: 1.0.0
 * Author: Axelerant
 * Author URI: https://axelerant.com
 * License: GPLv2 or later
 * Text Domain: wordpress-starter-premium
 * Domain Path: /languages
 */


/**
WordPress Starter Premium
Copyright (C) 2015 Axelerant

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License along
with this program; if not, write to the Free Software Foundation, Inc.,
51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

// Exit if accessed directly
if ( ! defined( 'ABSPATH' ) ) {
	exit;
}

define( 'WPSP_BASE', plugin_basename( __FILE__ ) );
define( 'WPSP_DIR', plugin_dir_path( __FILE__ ) );
define( 'WPSP_DIR_INC', WPSP_DIR . 'includes/' );
define( 'WPSP_DIR_LIB', WPSP_DIR_INC . 'libraries/' );
define( 'WPSP_NAME', 'WordPress Starter Premium' );
define( 'WPSP_PRODUCT_ID', 'TBD' );
define( 'WPSP_REQ_BASE', 'wordpress-starter/wordpress-starter.php' );
define( 'WPSP_REQ_NAME', 'WordPress Starter' );
define( 'WPSP_VERSION', '1.0.0' );

require_once WPSP_DIR_LIB . WPSP_REQ_BASE;
require_once WPSP_DIR_INC . 'requirements.php';

if ( ! wpsp_requirements_check() ) {
	return false;
}

require_once WPSP_DIR_INC . 'class-wordpress-starter-premium.php';


add_action( 'plugins_loaded', 'wordpress_starter_premium_init', 99 );


/**
 * @SuppressWarnings(PHPMD.LongVariable)
 * @SuppressWarnings(PHPMD.UnusedLocalVariable)
 */
function wordpress_starter_premium_init() {
	if ( ! is_admin() ) {
		return;
	}

	global $WPSP_Licensing;
	if ( is_null( $WPSP_Licensing ) ) {
		$WPSP_Licensing = new WordPress_Starter_Premium_Licensing();
	}

	if ( ! class_exists( 'EDD_SL_Plugin_Updater' ) ) {
		require_once WPSP_DIR_LIB . 'EDD_SL_Plugin_Updater.php';
	}

	$WPSP_Updater = new EDD_SL_Plugin_Updater(
		$WPSP_Licensing->store_url,
		__FILE__,
		array(
			'version' => WPSP_VERSION,
			'license' => $WPSP_Licensing->get_license(),
			'item_name' => WPSP_NAME,
			'author' => $WPSP_Licensing->author,
		)
	);

	if ( WordPress_Starter_Premium::version_check() ) {
		global $WordPress_Starter_Premium;
		if ( is_null( $WordPress_Starter_Premium ) ) {
			$WordPress_Starter_Premium = new WordPress_Starter_Premium();
		}

		do_action( 'wpsp_init' );
	}
}


register_activation_hook( __FILE__, array( 'WordPress_Starter_Premium', 'activation' ) );
register_deactivation_hook( __FILE__, array( 'WordPress_Starter_Premium', 'deactivation' ) );
register_uninstall_hook( __FILE__, array( 'WordPress_Starter_Premium', 'uninstall' ) );

?>


includes/class-wordpress-starter-premium-licensing.php
<?php
/**
WordPress Starter Premium
Copyright (C) 2015 Axelerant

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License along
with this program; if not, write to the Free Software Foundation, Inc.,
51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

require_once AIHR_DIR_INC . 'class-aihrus-licensing.php';

if ( class_exists( 'WordPress_Starter_Premium_Licensing' ) ) {
	return;
}


class WordPress_Starter_Premium_Licensing extends Aihrus_Licensing{
	public static $settings_id = WordPress_Starter_Settings::ID;


	public function __construct() {
		parent::__construct( WordPress_Starter_Premium::SLUG, WPSP_NAME );

		add_filter( 'wps_settings', array( __CLASS__, 'settings' ), 5 );
	}


	public function settings( $settings ) {
		$title = esc_html__( 'License Key for %1$s' );

		$settings[ WordPress_Starter_Premium::SLUG . 'license_key' ] = array(
			'section' => 'premium',
			'title' => esc_html__( 'License Key' ),
			'title' => sprintf( $title, WPSP_NAME ),
			'desc' => esc_html__( 'Required to enable premium plugin updating. Activation is automatic. Use `0` to deactivate.' ),
			'validate' => 'wpsp_update_license',
			'widget' => 0,
		);

		return $settings;
	}


}


/**
 *
 *
 * @SuppressWarnings(PHPMD.Superglobals)
 * @SuppressWarnings(PHPMD.LongVariable)
 */
function wpsp_update_license( $license ) {
	global $WPSP_Licensing;

	if ( ! empty( $_REQUEST['option_page'] ) && WordPress_Starter_Settings::ID == $_REQUEST['option_page'] ) {
		$current_license = $WPSP_Licensing->get_license();
		$valid_license   = $WPSP_Licensing->valid_license();
		if ( ! $valid_license || $license != $current_license ) {
			$result        = $WPSP_Licensing->update_license( $license );
			$valid_license = $WPSP_Licensing->valid_license();
			if ( ! $valid_license ) {
				WordPress_Starter_Premium::set_notice( 'notice_license', HOUR_IN_SECONDS );
			}
		}

		return $result;
	}

	return $license;
}


?>


includes/class-wordpress-starter-premium.php
<?php
/**
WordPress Starter Premium
Copyright (C) 2015 Axelerant

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License along
with this program; if not, write to the Free Software Foundation, Inc.,
51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

require_once AIHR_DIR_INC . 'class-aihrus-common.php';
require_once WPSP_DIR_INC . 'class-wordpress-starter-premium-licensing.php';

if ( class_exists( 'WordPress_Starter_Premium' ) ) {
	return;
}


class WordPress_Starter_Premium extends Aihrus_Common {
	const BASE    = WPSP_BASE;
	const ID      = 'wordpress-starter-premium';
	const SLUG    = 'wpsp_';
	const VERSION = WPSP_VERSION;

	public static $class = __CLASS__;
	public static $library_assets;
	public static $notice_key;
	public static $plugin_assets;


	public function __construct() {
		parent::__construct();

		self::$library_assets = plugins_url( '/includes/libraries/', dirname( __FILE__ ) );
		self::$library_assets = self::strip_protocol( self::$library_assets );

		self::$plugin_assets = plugins_url( '/assets/', dirname( __FILE__ ) );
		self::$plugin_assets = self::strip_protocol( self::$plugin_assets );

		add_action( 'admin_init', array( __CLASS__, 'admin_init' ) );
		add_action( 'init', array( __CLASS__, 'init' ) );
		add_shortcode( 'wordpress_starter_premium_shortcode', array( __CLASS__, 'wordpress_starter_premium_shortcode' ) );

		if ( ! WordPress_Starter::do_load() ) {
			return;
		}

		add_action( 'admin_menu', array( __CLASS__, 'admin_menu' ) );
	}


	/**
	 * @SuppressWarnings(PHPMD.LongVariable)
	 */
	public static function admin_init() {
		if ( ! self::version_check() ) {
			return;
		}

		global $WPSP_Licensing;
		if ( ! $WPSP_Licensing->valid_license() ) {
			self::set_notice( 'notice_license', DAY_IN_SECONDS );
		}

		add_filter( 'plugin_action_links', array( __CLASS__, 'plugin_action_links' ), 10, 2 );
	}


	public static function admin_menu() {
		add_action( 'admin_print_scripts', array( __CLASS__, 'scripts' ) );
		add_action( 'admin_print_styles', array( __CLASS__, 'styles' ) );
	}


	public static function init() {
		load_plugin_textdomain( self::ID, false, 'wordpress-starter-premium/languages' );

		if ( ! WordPress_Starter::do_load() ) {
			return;
		}

		self::load_options();

		add_action( 'wps_scripts', array( __CLASS__, 'scripts' ) );
		add_action( 'wps_styles', array( __CLASS__, 'styles' ) );
	}


	public static function plugin_action_links( $links, $file ) {
		if ( self::BASE == $file ) {
			array_unshift( $links, WordPress_Starter::$settings_link );
		}

		return $links;
	}


	public static function activation() {
		if ( ! current_user_can( 'activate_plugins' ) ) {
			return;
		}

		wps_init_options();
	}


	public static function deactivation() {
		if ( ! current_user_can( 'activate_plugins' ) ) {
			return;
		}
	}


	/**
	 * @SuppressWarnings(PHPMD.LongVariable)
	 */
	public static function uninstall() {
		if ( ! current_user_can( 'activate_plugins' ) ) {
			return;
		}

		require_once WPSP_DIR_INC . 'class-wordpress-starter-premium-licensing.php';

		$WPSP_Licensing = new WordPress_Starter_Premium_Licensing();
		$WPSP_Licensing->deactivate_license();
	}


	/**
	 * @SuppressWarnings(PHPMD.UnusedFormalParameter)
	 */
	public static function scripts( $atts ) {
		do_action( 'wpsp_scripts' );
	}


	public static function styles() {
		if ( ! is_admin() ) {
			wp_register_style( __CLASS__, self::$plugin_assets . 'css/wordpress-starter-premium.css' );
			wp_enqueue_style( __CLASS__ );
		}

		do_action( 'wpsp_styles' );
	}


	public static function wordpress_starter_premium_shortcode( $atts ) {
		WordPress_Starter::call_scripts_styles( $atts );

		return __CLASS__ . ' shortcode';
	}


	public static function version_check() {
		$valid_version = true;

		if ( ! $valid_version ) {
			$deactivate_reason = esc_html__( 'Failed version check' );
			aihr_deactivate_plugin( self::BASE, WPSP_NAME, $deactivate_reason );
			self::check_notices();
		}

		return $valid_version;
	}


	public static function load_options() {
		add_filter( 'wps_sections', array( __CLASS__, 'sections' ) );
		add_filter( 'wps_settings', array( __CLASS__, 'settings' ) );
	}


	public static function sections( $sections ) {
		$sections['premium'] = esc_html__( 'Premium' );

		return $sections;
	}


	public static function settings( $settings ) {
		$settings['disable_donate'] = array(
			'section' => 'premium',
			'title' => esc_html__( 'Disable Donate Text?' ),
			'desc' => esc_html__( 'Remove "If you like…" text with the donate and premium purchase links from the settings screen.' ),
			'type' => 'checkbox',
		);

		return $settings;
	}


	public static function notice_license() {
		$post_type     = null;
		$settings_id   = WordPress_Starter_Settings::ID;
		$required_name = WPSP_REQ_NAME;
		$purchase_url  = 'https://store.axelerant.com/downloads/wordpress-starter-premium-wordpress-plugin/';
		$item_name     = WPSP_NAME;
		$product_id    = WPSP_PRODUCT_ID;
		$license       = wps_get_option( WordPress_Starter_Premium::SLUG . 'license_key' );

		aihr_notice_license( $post_type, $settings_id, $required_name, $purchase_url, $item_name, $product_id, $license );
	}
}

?>
